app/graphql/resolvers/settings_resolver.rb

```diff
@@ -0,0 +1,21 @@
+module Resolvers
+  class SettingsResolver < BaseResolver
+    type [Types::Setting], null: false
+
+    argument :search, String, 'Search query', required: false
+
+    def resolve(search: nil)
+      return [] unless user&.can?(:view_settings)
+
+      scope = Foreman.settings
+      scope.search_for(search) if search
```

Comment:
Searching currently does nothing
```suggestion
      scope = scope.search_for(search) if search
```

Reply:
:facepalm: Thanks